public/js/modal.js

```diff
@@ -0,0 +1,46 @@
+document.addEventListener("DOMContentLoaded", () => {
+  const addIngredientButton = document.querySelector("#add-ingredient-btn");
+  const ingredientForm = document.querySelector("#recipe-form");
+  let ingredientCount = 2;
+
+  addIngredientButton.addEventListener("click", () => {
+    console.log("The add ingredient button has been clicked!");
+    const input = document.createElement("input");
+    input.type = "text";
+    input.name = `ingredients[]`;
+    input.placeholder = `ingredient ${ingredientCount}`;
+    input.classList.add(
+      "input",
+      "input-bordered",
+      "w-full",
+      "max-w-xs",
+      "bg-white",
+      "text-dark-slate-gray",
+      "my-2"
+    );
+    ingredientForm.insertBefore(input, addIngredientButton);
+    ingredientCount++;
+  });
+
+  const addStepsButton = document.querySelector("#add-steps-btn");
+  const directionsForm = document.querySelector("#directions-form");
+  let stepsCount = 2;
+
+  addStepsButton.addEventListener("click", () => {
+    console.log("The add ingredient button has been clicked!");
```

Comment:
Change this console log message to reflect that we're add a step/direction, not an ingredient.